homeassistant/components/hue/sensor.py

```diff
@@ -0,0 +1,324 @@
+"""Support for the Philips Hue sensors."""
+import asyncio
+from datetime import timedelta
+import logging
+from time import monotonic
+
+import async_timeout
+
+from homeassistant.components import hue
+from homeassistant.components.binary_sensor import (
+    BinarySensorDevice, ENTITY_ID_FORMAT as BINARY_ENTITY_ID_FORMAT)
```

Comment:
This is not allowed. Move to `hue/binary_sensor.py`

Reply:
This isn't something that is enforced elsewhere and it's going to make for untidy and non-obvious code when different entities that are updated through the same mechanism and are part of the same device are split across two files.

What degree of separation would you like to see here?

1. Move the binary sensor entity itself to a separate file and import it here?
2. Duplicate all this code to the binary sensor module?
3. Split the shared code into a different module, and keep the sensor-specific content here, and binary sensor content in a different module?

Reply:
The homekit integration for example, bundles sensors and binary sensors in the same module.

Reply:
This is enforced everywhere else. If you find a place where it is not done, that should be fixed. Each platform for another entity component is its own file.

Reply:
Tried to implement this in the latest update, though it's kind of ugly. I tried to keep the single API call to update all sensors, binary or not. This whole dichotomy of binary sensors and gauges doesn't really fit the reality neatly, here.